### PR TITLE
cmd/tailscale/cli: move tskey-wrap functionality under lock sign

### PR DESCRIPTION
**Pull request by twitchyliquid64**

### CODE REVIEW

File: cmd/tailscale/cli/network-lock.go [hunk 1]
```diff
@@ -427,13 +435,19 @@ func runNetworkLockModify(ctx context.Context, addArgs, removeArgs []string) err
 
 var nlSignCmd = &ffcli.Command{
 	Name:       "sign",
-	ShortUsage: "sign <node-key> [<rotation-key>]",
-	ShortHelp:  "Signs a node key and transmits the signature to the coordination server",
-	LongHelp:   "Signs a node key and transmits the signature to the coordination server",
-	Exec:       runNetworkLockSign,
+	ShortUsage: "sign <node-key> [<rotation-key>] or sign <preauth-key>",
+	ShortHelp:  "Signs a node or preauth-key",
```
Comment (mayakacz) [inline cmd/tailscale/cli/network-lock.go:439]:
```suggestion
	ShortHelp:  "Signs a node or pre-approved auth key",
```

File: cmd/tailscale/cli/network-lock.go [hunk 2]
```diff
@@ -427,13 +435,19 @@ func runNetworkLockModify(ctx context.Context, addArgs, removeArgs []string) err
 
 var nlSignCmd = &ffcli.Command{
 	Name:       "sign",
-	ShortUsage: "sign <node-key> [<rotation-key>]",
-	ShortHelp:  "Signs a node key and transmits the signature to the coordination server",
-	LongHelp:   "Signs a node key and transmits the signature to the coordination server",
-	Exec:       runNetworkLockSign,
+	ShortUsage: "sign <node-key> [<rotation-key>] or sign <preauth-key>",
```
Comment (mayakacz) [inline cmd/tailscale/cli/network-lock.go:438]:
```suggestion
	ShortUsage: "sign <node-key> [<rotation-key>] or sign <auth-key>",
```

File: cmd/tailscale/cli/network-lock.go [hunk 3]
```diff
@@ -427,13 +435,19 @@ func runNetworkLockModify(ctx context.Context, addArgs, removeArgs []string) err
...
+	ShortHelp:  "Signs a node or preauth-key",
+	LongHelp: `Either:
+  - signs a node key and transmits the signature to the coordination server, or
+  - signs a pre-auth key, printing it in a form that can be used to bring up nodes under tailnet lock`,
```
Comment (mayakacz) [inline cmd/tailscale/cli/network-lock.go:442]:
```suggestion
  - signs a pre-approved auth key, printing it in a form that can be used to bring up nodes under tailnet lock`,
```

**Review by mayakacz — APPROVED** [review]
thanks!

File: cmd/tailscale/cli/network-lock.go [hunk 4]
```diff
@@ -657,21 +663,25 @@ func runTskeyWrapCmd(ctx context.Context, args []string) error {
...
 	}
-	if strings.HasPrefix(args[0], "tskey-auth-") && strings.Index(args[0][len("tskey-auth-"):], "-") > 0 {
+	if strings.HasPrefix(keyStr, "tskey-auth-") && strings.Index(keyStr[len("tskey-auth-"):], "-") > 0 {
 		// We don't want to accidentally embed the nonce part of the authkey in
 		// the event the format changes. As such, we make sure its in the format we
 		// expect (tskey-auth-<stableID, inc CNTRL suffix>-nonce) before we parse
 		// out and embed the stableID.
-		s := strings.TrimPrefix(args[0], "tskey-auth-")
+		s := strings.TrimPrefix(keyStr, "tskey-auth-")
 		m["authkey_stableid"] = s[:strings.Index(s, "-")]
```
[the remark this answers is not in the document]
Comment (twitchyliquid64) [reply cmd/tailscale/cli/network-lock.go:685]: Its guarded by the check in the earlier if statement

**Pull request merged**